src/time/SDL_time.c

```diff
@@ -0,0 +1,225 @@
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+
+#include "SDL_time_c.h"
+
+static SDL_bool time_initialized;
+
+/* The following algorithms are based on those of Howard Hinnant and are in the public domain.
+ *
+ * http://howardhinnant.github.io/date_algorithms.html
+ */
+
+/* Given a calendar date, returns days since Jan 1 1970, and optionally
+ * the day of the week (0-6, 0 is Sunday) and day of the year (0-365).
+ */
+Sint64 SDL_CivilToDays(int year, int month, int day, int *day_of_week, int *day_of_year)
+{
+
+    year -= month <= 2;
+    const int era = (year >= 0 ? year : year - 399) / 400;
+    const unsigned yoe = (unsigned)(year - era * 400);                                  // [0, 399]
+    const unsigned doy = (153 * (month > 2 ? month - 3 : month + 9) + 2) / 5 + day - 1; // [0, 365]
+    const unsigned doe = yoe * 365 + yoe / 4 - yoe / 100 + doy;                         // [0, 146096]
+    const Sint64 z = (Sint64)(era) * 146097 + (Sint64)(doe)-719468;
+
+    if (day_of_week) {
+        *day_of_week = (int)(z >= -4 ? (z + 4) % 7 : (z + 5) % 7 + 6);
+    }
+    if (day_of_year) {
+        /* This algorithm considers March 1 to be the first day of the year, so offset by Jan + Feb. */
+        if (doy > 305) {
+            /* Day 0 is the first day of the year. */
+            *day_of_year = doy - 306;
+        } else {
+            const int doy_offset = 59 + (!(year % 4) && ((year % 100) || !(year % 400)));
+            *day_of_year = doy + doy_offset;
+        }
+    }
+
+    return z;
+}
+
+void SDL_InitTime()
+{
+    if (time_initialized) {
+        return;
+    }
+    /* Default to ISO 8061 date format, as it is unambiguous, and 24 hour time. */
+    SDL_DATE_FORMAT df = SDL_DATE_FORMAT_YYYYMMDD;
+    SDL_TIME_FORMAT tf = SDL_TIME_FORMAT_24HR;
+    SDL_PropertiesID props = SDL_GetGlobalProperties();
+
+    SDL_GetSystemTimeLocalePreferences(&df, &tf);
+
+    SDL_SetNumberProperty(props, SDL_PROP_GLOBAL_SYSTEM_DATE_FORMAT_NUMBER, df);
+    SDL_SetNumberProperty(props, SDL_PROP_GLOBAL_SYSTEM_TIME_FORMAT_NUMBER, tf);
```

Comment:
Maybe check to see if these properties were already set first?

Reply:
Also, instead of setting these here, you might want to use the df/tf values as defaults when you query the properties, in case the system settings change over the lifetime of the application.

Reply:
They are now checked for having previous values set before initializing with the system defaults.